olive/passes/onnx/model_builder.py

```diff
@@ -150,6 +168,29 @@ def _run_for_config(
             if not isinstance(model, PyTorchModelHandler):
                 raise ValueError("metadata_only has to be true with ONNXModel as input.")
 
+        merged_model_path = None
```

Comment:
Is this feature requested by some customer? If this is needed, I think it is better if we provide a separate pass to merge the adapter weights. Otherwise, we are adding similar merge options and functionalities to multiple passes like onnxconversion and modelbuilder when these are technically not related to the pass. 
If this is done, we could also remove the merge weights options from onnx conversion pass.

Reply:
Yes, there is a offline session will finetune phi3 and run genai.
But not urgent, let me update it with seprated pass.

Reply:
Updated.
But for conversion, since the quantized_modules need to be unload before run `merge_and_unload`. I did not remove the logics directly. How about update it with separated PR.